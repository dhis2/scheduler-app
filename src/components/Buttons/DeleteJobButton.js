import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Button } from '@dhis2/ui'
import i18n from '@dhis2/d2-i18n'
import { DeleteJobModal } from '../Modal'

const DeleteJobButton = ({ id, onSuccess }) => {
    const [showModal, setShowModal] = useState(false)

    return (
        <React.Fragment>
            <Button destructive onClick={() => setShowModal(true)}>
                {i18n.t('Delete job')}
            </Button>
            {showModal && (
                <DeleteJobModal
                    id={id}
                    hideModal={
                        /* istanbul ignore next */
                        () => setShowModal(false)
                    }
                    onSuccess={onSuccess}
                />
            )}
        </React.Fragment>
    )
}

const { string, func } = PropTypes

DeleteJobButton.propTypes = {
    id: string.isRequired,
    onSuccess: func.isRequired,
}

export default DeleteJobButton
