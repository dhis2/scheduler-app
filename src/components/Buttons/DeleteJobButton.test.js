import React from 'react'
import { shallow, mount } from 'enzyme'
import DeleteJobButton from './DeleteJobButton'

describe('<DeleteJobButton>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<DeleteJobButton id="1" />)

        expect(wrapper).toMatchSnapshot()
    })

    it('shows the modal when button is clicked', () => {
        const wrapper = mount(<DeleteJobButton id="id" />)

        expect(wrapper.find('DeleteJobModal')).toHaveLength(0)

        wrapper.find('button').simulate('click')

        expect(wrapper.find('DeleteJobModal')).toHaveLength(1)
    })
})
