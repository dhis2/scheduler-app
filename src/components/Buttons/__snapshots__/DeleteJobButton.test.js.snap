// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteJobButton> renders correctly 1`] = `
<Fragment>
  <Button
    dataTest="dhis2-uicore-button"
    destructive={true}
    onClick={[Function]}
    type="button"
  >
    Delete
  </Button>
</Fragment>
`;
